mlir/test/Dialect/EmitC/invalid_types.mlir

```diff
@@ -14,6 +14,34 @@ func.func @illegal_opaque_type_2() {
 
 // -----
 
+func.func @illegal_opaque_type() {
```

Comment:
nit: Using the opaque type as function argument instead of type for emitc.variable might make the tests a bit easier to read. I was first confused by the second emitc.opaque on the same line

Reply:
I was following an unspoken "convention". Happy to modify to use function arguments. I also missed the return type.